background.ts

```diff
@@ -749,17 +764,121 @@ function newK8sManager() {
 
 /**
  * Implement the methods the HttpCommandServer needs to service its requests.
- * These methods should be thin wrappers around existing functionality in the rest of the backend.
- * Getters normally return strings, either JSON strings or scalars.
- * Setters normally return either `true | false`, or possibly `true | error`.
- * The `requestShutdown` is a special case that never returns.
+ * These methods do two things:
+ * 1. Verify the semantics of the parameters (the server just checks syntax).
+ * 2. Provide a thin wrapper over existing functionality in this module.
+ * Getters, on success, return status 200 and a string that may be JSON or simple.
+ * Setters, on success, return status 202, possibly with a human-readable status note.
+ * The `requestShutdown` method is a special case that never returns.
  */
 class BackgroundCommandWorker implements CommandWorkerInterface {
   getSettings() {
     return JSON.stringify(cfg, undefined, 2);
   }
 
-  requestShutdown() {
+  /**
+   * On failure, returns status 400, with a human-readable error message.
+   * @param newSettings
+   */
+  async updateSettings(newSettings: UpdatableSettings): Promise<[string, string]> {
```

Comment:
Actually, this was in my prototype, but Jan wanted something friendlier, and that well also lend itself well to processing the values on the command-line. And it was only in my prototype because it was easy to implement (although harder to use).

It's also possible for the CLI to translate the received JSON into something "friendlier" like tabular format. But harder to convert tabular format into JSON if the user is scripting the app. That's the next step, where I might add a `--output=json|table` option to do just that.

Reply:
Yeah, I'm happy with having either the simplified mapping or the whole raw settings, _as long as they match_. I think it'd be okay for the wire format to be the simplified form, if you think that's better (or the other way around).

Reply:
Updated this so `curl /set -d "$(curl /list-settings)"` is a no-op